build-tools/packages/build-infrastructure/src/fluidRepo.ts

```diff
@@ -235,3 +237,37 @@ export function getAllDependenciesInRepo(
 		workspaces: [...workspaces],
 	};
 }
+
+export async function setDependencyRange(
+	packagesToUpdate: IPackage[],
+	dependencies: IPackage[],
+	dependencyRange: InterdependencyRange,
+): Promise<void> {
+	const dependencySet = new Set(dependencies.map((d) => d.name));
+	// collect the "save" promises to resolve in parallel
+	const savePromises: Promise<void>[] = [];
+
+	for (const pkg of packagesToUpdate) {
+		for (const { name: depName, depKind } of pkg.combinedDependencies) {
+			if (dependencySet.has(depName)) {
+				const depRange =
+					typeof dependencyRange === "string"
+						? dependencyRange
+						: dependencyRange instanceof semver.SemVer
+							? dependencyRange.version
+							: undefined;
+
+				// Update the version in packageJson
+				if (depKind === "prod" && pkg.packageJson.dependencies) {
+					pkg.packageJson.dependencies[depName] = depRange;
+				} else if (depKind === "dev" && pkg.packageJson.devDependencies) {
+					pkg.packageJson.devDependencies[depName] = depRange;
+				} else if (depKind === "peer" && pkg.packageJson.peerDependencies) {
+					pkg.packageJson.peerDependencies[depName] = depRange;
+				}
```

Comment:
It's a shame that TypeScript can't infer that those properties exist, since the "kind" implies that the value is there. But it also kind of makes sense why it can't infer that - there's no obvious connection between the string "dev" and the devDependencies property in package.json.

I think there are two options, one is what you've done. I would make the condition explict though - pkg.packageJson.dependencies !== undefined.

The other option is to assert that it's defined. Something like `assert(pkg.packageJson.dependencies !== undefined, "mesage")` or pkg.packageJson.dependencies![key] - but the latter requires a lint disable. I think what you've done (with explicit checks) is probably best.